frontend/src/pages/modelServing/screens/projects/ModelServingPlatform.tsx

```diff
@@ -0,0 +1,156 @@
+    disableModelMesh,
+  );
+
+  const isModelMesh =
+    currentProjectServingPlatform === ServingRuntimePlatform.MULTI ||
+    platformSelection === ServingRuntimePlatform.MULTI;
+
+  const shouldShowPlatformSelection =
+    !disableKServe && !disableModelMesh && !currentProjectServingPlatform;
+
+  const initialState = !platformSelection && !currentProjectServingPlatform;
+
+  return (
+    <>
+      <DetailsSection
+        id={ProjectSectionID.MODEL_SERVER}
+        title={ProjectSectionTitles[ProjectSectionID.MODEL_SERVER]}
+        actions={
+          initialState
+            ? undefined
+            : [
+                <ModelServingPlatformButtonAction
+                  tooltipContent={
+                    // show tooltip for deploying a model
+                    emptyTemplates
+                      ? `At least one serving runtime must be enabled to ${
+                          isModelMesh ? 'add a model server' : 'deploy a model'
+                        }. Contact your administrator.`
+                      : isModelMesh
+                      ? 'A model server specifies resources available for use by one or more supported models, and includes a serving runtime.'
+                      : ''
+                  }
+                  isLoading={!templatesLoaded}
+                  isDisabled={!templatesLoaded || emptyTemplates}
+                  onClick={() => {
+                    if (isModelMesh) {
+                      setOpen(true);
+                    }
+                    // else, show the kserve deploy model modal
+                  }}
+                  key="serving-runtime-actions"
+                  title={isModelMesh ? 'Add model server' : 'Deploy model'}
+                />,
+              ]
+        }
+        isLoading={!servingRuntimesLoaded && !templatesLoaded}
+        isEmpty={!shouldShowPlatformSelection && emptyModelServer}
+        loadError={servingRuntimeError || templateError}
+        emptyState={
+          <EmptyDetailsList
+            title={isModelMesh ? 'No model servers' : 'No deployed models'}
+            description={
+              isModelMesh
+                ? 'Before deploying a model, you must first add a model server.'
+                : 'To get started, deploy a model.'
+            }
+            icon={PlusCircleIcon}
+          />
+        }
+        labels={
+          currentProjectServingPlatform && [
+            <Label key="serving-platform-label">
+              {isModelMesh ? 'Multi-model serving enabled' : 'Single model serving enabled'}
+            </Label>,
+          ]
+        }
+      >
+        {shouldShowPlatformSelection ? (
+          <ModelServingPlatformSelect
+            platform={platformSelection}
+            setPlatform={setPlatformSelection}
+          />
+        ) : (
+          // show kserve table if `isModelMesh` is false
+          <ModelMeshServingRuntimeTable />
+        )}
+      </DetailsSection>
+      <ManageServingRuntimeModal
```

Comment:
ok, as we talked, I'll do the follow up here, and conditional render the modal depending on the selection made.